http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceV2InterpSpec.scala

```diff
@@ -194,6 +197,65 @@ class RuntimeServiceV2InterpSpec extends AnyFlatSpec with LeonardoTestSuite with
     exc shouldBe a[RuntimeAlreadyExistsException]
   }
 
+  it should "fail to create a runtime with existing disk if there are multiple disks" in isolatedDbTest {
+    val res = for {
+
+      disk1 <- makePersistentDisk(Some(DiskName("d1")), cloudContextOpt = Some(cloudContextAzure)).save()
+      disk2 <- makePersistentDisk(Some(DiskName("d2")), cloudContextOpt = Some(cloudContextAzure)).save()
+
+      err <- runtimeV2Service
+        .createRuntime(userInfo, name0, workspaceId, true, defaultCreateAzureRuntimeReq)
+        .attempt
+    } yield err should (be(Left(MultiplePersistentDisksException(workspaceId, 2, List(disk2, disk1)))) or
+      be(Left(MultiplePersistentDisksException(workspaceId, 2, List(disk1, disk2)))))
+    res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
+  it should "fail to create a runtime with existing disk if there are 0 disks" in isolatedDbTest {
+    val exc = runtimeV2Service
+      .createRuntime(userInfo, name0, workspaceId, true, defaultCreateAzureRuntimeReq)
+      .attempt
+      .unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+      .swap
+      .toOption
+      .get
+    exc shouldBe a[NoPersistentDiskException]
+  }
+
+  it should "fail to create a runtime with existing disk if disk isn't ready" in isolatedDbTest {
+    val res = for {
+
+      disk <- makePersistentDisk(Some(DiskName("d1")), cloudContextOpt = Some(cloudContextAzure)).save()
+      now <- IO.realTimeInstant
+      _ <- persistentDiskQuery.updateStatus(disk.id, DiskStatus.Restoring, now).transaction
+
+      err <- runtimeV2Service
+        .createRuntime(userInfo, name0, workspaceId, true, defaultCreateAzureRuntimeReq)
+        .attempt
+    } yield err shouldBe Left(PersistentDiskNotReadyException(disk.id, DiskStatus.Restoring))
+    res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
+  it should "fail to create a runtime with existing disk if disk is attached to non-deleted runtime" in isolatedDbTest {
+    val res = for {
+      _ <- runtimeV2Service
+        .createRuntime(userInfo, name0, workspaceId, false, defaultCreateAzureRuntimeReq)
+
+      disks <- DiskServiceDbQueries
+        .listDisks(Map.empty, includeDeleted = false, Some(userInfo.userEmail), Some(cloudContextAzure))
+        .transaction
+      disk = disks.head
+      now <- IO.realTimeInstant
+      _ <- persistentDiskQuery.updateStatus(disk.id, DiskStatus.Ready, now).transaction
+
+      err <- runtimeV2Service
+        .createRuntime(userInfo, name1, workspaceId, true, defaultCreateAzureRuntimeReq)
+        .attempt
+
+    } yield err shouldBe Left(NoUnattachedPersistentDiskException(disk.id, name0))
+    res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
```

Comment:
I'd like to see a test that ensures `createRuntime` behaves properly if called twice in a row with `useExistingDisk = true`.

I.e., the first time it should succeed and attach the second disk. The second time it should fail, as that existing disk is now attached. It's a bit redundant, but I think its a common enough use case to smash 'send' twice in a row with the same request to ensure we are doing the right thing.